docs/proposals/17-shoot-control-plane-migration-bad-case.md

```diff
@@ -0,0 +1,102 @@
+
+For example, for a shoot named `i500152-gcp` in project `dev` that uses an internal domain suffix `internal.dev.k8s.ondemand.com` and is scheduled on a seed with an identity `shoot--i500152--gcp2-0841c87f-8db9-4d04-a603-35570da6341f-sap-landscape-dev`, the owner DNS record is a TXT record with a domain name `owner.i500152-gcp.dev.internal.dev.k8s.ondemand.com` and a single value `shoot--i500152--gcp2-0841c87f-8db9-4d04-a603-35570da6341f-sap-landscape-dev`. The owner DNS record is created and maintained by reconciling an `owner` DNSRecord resource, if the recently introduced DNSRecords feature is enabled via the `UseDNSRecords` feature gate. 
+
+Unlike other extension resources, the `owner` DNSRecord resource is not reconciled every time the shoot is reconciled, but only when the resource is created. Therefore, the owner DNS record value (the owner ID) is updated only when the shoot is migrated to a different seed. For more information, see [Add handling of owner DNSRecord resources](https://github.com/gardener/gardener/pull/4307).
+
+The owner DNS record domain name and owner ID are passed to components that need to perform ownership checks, such as the `backup-restore` container of the `etcd-main` StatefulSet, and all extension controllers. These components then check regularly whether the actual owner ID (the value of the record) matches the passed ID. If they don't, the ownership check is considered failed, which causes the special behavior described below.
+
+**Note:** A previous revision of this document proposed using "sync objects" written to and read from the backup container of the source seed as JSON files by the `etcd-backup-restore` processes in both seeds. With the introduction of owner DNS records such sync objects are no longer needed.
+
+For the destination seed to actually become the owner, it needs to acquire the shoot's etcd data by copying the final full snapshot (and potentially also older snapshots) from the backup container of the source seed.
+
+The mechanism to copy the snapshots and pass the ownership from the source to the destination seed consists of the following steps:
+
+1. The reconciliation flow ("restore" phase) is triggered in the destination seed without first executing the migration flow in the source seed (or perhaps it was executed, but it failed, and its state is currently unknown).
+
+2. The `owner` DNSRecord resource is created in the destination seed. As a result, the actual owner DNS record is updated with the destination seed ID. From this point, ownership checks by the `etcd-backup-restore` process and [extension controller watchdogs](#extension-controller-watchdogs) in the source seed will fail, which will cause the special behavior described below. 
```

Comment:
Not immediately, right? Only after a certain TTL?

Reply:
While testing the PoC we never observed any delays comparable to the owner DNS record TTL (which was set to 2 minutes). The source seed components were performing the ownership check every 30 seconds, and most often the first ownership check after the DNS record change already failed (so the mean time for the components to notice the change was about 15 seconds). This behavior is perhaps not guaranteed, but at least likely. In the worst case, there might be a delay for the TTL duration or perhaps even longer, but this doesn't change anything except making the restoration flow take a bit longer to execute.